mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -171,6 +180,31 @@ static cl::opt<int> blockSize("block_size", cl::desc("Block size"),
 static cl::opt<int> gridSize("grid_size", cl::desc("Grid size"),
                              cl::value_desc("Grid size"), cl::init(0));
 
+int getDeviceId() // Get default device
+{
+  int device = 0;
+  auto status = hipGetDevice(&device);
```

Comment:
as a test tool, the output of `mlir-miopen-driver` should completely be controlled by users, NOT by the system characteristic. Therefore I'm not fond of getting these 2 values queried and set at `mlir-miopen-driver`.

Reply:
Thanks for the feedback. Thinking out loud here:

- For cpp backend (when `mlir-rocm-runner` is not involved), it needs to pull in the `cu` and `arch` somewhere. I think it should be fine to rely only on user input.
- For ISA backend, there needs to be another callback function (like the existing `launchDimCallback()`), where the `mlir-rocm-runner` is able to inserts the `arch` and `cu` information back to `ConvolutionContext` (In the `AffixTuningParameters` step), just so that the correct tuning parameters can be picked.

If you'd be fine with this design, I amend the PR this way.

Reply:
this sounds like a better approach to me, thanks!